source/extensions/filters/listener/original_dst/config.cc

```diff
@@ -22,10 +22,28 @@ class OriginalDstConfigFactory : public Server::Configuration::NamedListenerFilt
   Network::ListenerFilterFactoryCb createListenerFilterFactoryFromProto(
       const Protobuf::Message&,
       const Network::ListenerFilterMatcherSharedPtr& listener_filter_matcher,
-      Server::Configuration::ListenerFactoryContext&) override {
-    return [listener_filter_matcher](Network::ListenerFilterManager& filter_manager) -> void {
+      Server::Configuration::ListenerFactoryContext& context) override {
+#ifdef WIN32
+    // On Windows we need to do some extra validation for the Original Destination filter.
+    // In particular we need to check if:
+    // 1. The platform supports the original destination feature
+    // 2. The `traffic_direction` property is set on the listener. This is required to redirect the
+    // traffic.
+    if (context.listenerConfig().direction() == envoy::config::core::v3::UNSPECIFIED) {
+      throw EnvoyException("[Windows] Setting original destination filter on a listener without "
+                           "specifying the traffic_direction."
+                           "Configure the traffic_direction listener option");
+    }
+    if (!Network::win32SupportsOriginalDestination()) {
+      throw EnvoyException("[Windows] Envoy was compiled without support for `SO_ORIGINAL_DST`, "
+                           "the original destination filter can not be used");
+    }
```

Comment:
cant not - > cannot

Reply:
fixed in c6718e872a64a57e440c090fa2df9fc6dd5a40eb